sentry/src/main/java/io/sentry/util/TracingUtils.java

```diff
@@ -0,0 +1,109 @@
+package io.sentry.util;
+
+import io.sentry.Baggage;
+import io.sentry.BaggageHeader;
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.PropagationContext;
+import io.sentry.Scope;
+import io.sentry.SentryOptions;
+import io.sentry.SentryTraceHeader;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class TracingUtils {
+
+  public static void startNewTrace(final @NotNull IHub hub) {
+    hub.configureScope(
+        scope -> {
+          scope.setPropagationContext(new PropagationContext());
+        });
+  }
+
+  public static @Nullable TracingHeaders traceIfAllowed(
+      final @NotNull IHub hub,
+      final @NotNull String requestUrl,
+      @Nullable List<String> thirdPartyBaggageHeaders,
+      final @Nullable ISpan span) {
+    final @NotNull SentryOptions sentryOptions = hub.getOptions();
+    if (sentryOptions.isTraceSampling() && isAllowedToSendTo(requestUrl, sentryOptions)) {
+      return trace(hub, thirdPartyBaggageHeaders, span);
+    }
+
+    return null;
+  }
+
+  public static @Nullable TracingHeaders trace(
+      final @NotNull IHub hub,
+      @Nullable List<String> thirdPartyBaggageHeaders,
+      final @Nullable ISpan span) {
+    final @NotNull SentryOptions sentryOptions = hub.getOptions();
+
+    if (span != null && !span.isNoOp()) {
+      return new TracingHeaders(
+          span.toSentryTrace(), span.toBaggageHeader(thirdPartyBaggageHeaders));
+    } else {
+      final @NotNull AtomicReference<TracingHeaders> returnValue = new AtomicReference<>();
+      hub.configureScope(
+          (scope) -> {
+            maybeUpdateBaggage(scope, sentryOptions);
+            final @NotNull PropagationContext propagationContext = scope.getPropagationContext();
+
+            final @Nullable Baggage baggage = propagationContext.getBaggage();
+            @Nullable BaggageHeader baggageHeader = null;
+            if (baggage != null) {
+              baggageHeader =
+                  BaggageHeader.fromBaggageAndOutgoingHeader(baggage, thirdPartyBaggageHeaders);
+            }
+
+            returnValue.set(
+                new TracingHeaders(
+                    new SentryTraceHeader(
+                        propagationContext.getTraceId(), propagationContext.getSpanId(), null),
+                    baggageHeader));
+          });
+      return returnValue.get();
+    }
+  }
+
+  public static void maybeUpdateBaggage(
```

Comment:
Not an expert around this area but would it make sense to return a `boolean` here, indicating if the baggage update was successful? `trace()` could then check for the return value and only create the tracing headers if the baggage update succeeded.

Reply:
Not sure I follow. We'd still want to add the headers even if there was no update to `baggage`.